RxTodo/Sources/ViewControllers/TaskListViewModel.swift

```diff
@@ -13,103 +13,116 @@ import RxSwift
-      .map { indexPath in
-        let task = tasks.value[indexPath.row]
-        return TaskEditViewModel(mode: .edit(task))
-      }
-
-    self.presentTaskEditViewModel = Observable.of(presentAddViewModel, presentEditViewModel).merge()
-      .asDriver(onErrorDriveWith: .empty())
-
-    //
-    // Model Service
-    //
-    Task.didCreate
-      .subscribe(onNext: { task in
-        tasks.value.insert(task, at: 0)
-      })
-      .addDisposableTo(self.disposeBag)
-
-    Task.didUpdate
-      .subscribe(onNext: { task in
-        if let index = tasks.value.index(of: task) {
-          tasks.value[index] = task
-        }
-      })
-      .addDisposableTo(self.disposeBag)
-
-    Task.didDelete
-      .subscribe(onNext: { task in
-        if let index = tasks.value.index(of: task) {
-          tasks.value.remove(at: index)
-        }
-      })
-      .addDisposableTo(self.disposeBag)
-  }
-
+    // MARK: Input
+
+    let addButtonDidTap = PublishSubject<Void>()
+    let itemDidSelect = PublishSubject<IndexPath>()
+    var itemDeleted = PublishSubject<IndexPath>()
+
+
+    // MARK: Output
+
+    let navigationBarTitle: Driver<String?>
+    let sections: Driver<[TaskListSection]>
+    let presentTaskEditViewModel: Driver<TaskEditViewModelType>
+
+
+    // MARK: Private
+
+    private let disposeBag = DisposeBag()
+    private var tasks: Variable<[Task]>
+
+    init() {
+        let defaultTasks = [
+            Task(title: "Go to https://github.com/devxoul"),
+            Task(title: "Star repositories I am intersted in"),
+            Task(title: "Make a pull request"),
+        ]
+
+        tasks = Variable<[Task]>(defaultTasks)
+
+        navigationBarTitle = .just("Tasks")
+
+        sections = tasks.asObservable()
+            .map { tasks in
+                let cellModels = tasks.map(TaskCellModel.init)
+                let section = TaskListSection(model: Void(), items: cellModels)
+                return [section]
+            }
+            .asDriver(onErrorJustReturn: [])
+
+        itemDeleted
+            .withLatestFrom(tasks.asObservable()) { $0 }
+            .subscribe(onNext: { indexPath, tasks in
+                let task = tasks[indexPath.row]
+                Task.didDelete.onNext(task)
+            })
+            .addDisposableTo(disposeBag)
+
+        // View Controller Navigations
+        let presentAddViewModel: Observable<TaskEditViewModelType> = addButtonDidTap
+            .map { TaskEditViewModel(mode: .new) }
+
+        let presentEditViewModel: Observable<TaskEditViewModelType> = itemDidSelect
+            .withLatestFrom(tasks.asObservable()) { $0 }
+            .map { indexPath, tasks in
+                let task = tasks[indexPath.row]
+                return TaskEditViewModel(mode: .edit(task))
+            }
+
+        presentTaskEditViewModel = Observable.of(presentAddViewModel, presentEditViewModel).merge()
+            .asDriver(onErrorDriveWith: .empty())
+
+        // Model Service
+        Task.didCreate
```

Comment:
In this model service section I made a small change that since it is not good to modify the value of `tasks: Variable<[Task]>` directly, I used `withLatestFrom` and modified by using map operator, then bindTo original tasks.